src/ShorSimulator.cpp

```diff
@@ -119,7 +117,7 @@ std::map<std::string, unsigned int> ShorSimulator::Simulate(unsigned int shots)
         }
 
         if (approximate && (i + 1) % mod == 0) {
-            fidelity *= ApproximateByFidelity(0.9, false, true);
+            final_fidelity *= ApproximateByFidelity(step_fidelity, false, true);
```

Comment:
for some reason this line is not covered by any tests. could you add a unit test for this?

Reply:
Good catch, I'll a test.